### ISSUE TITLE
ERROR   : Failed to download checksum for the file 'Updates.xml' from mirrors '['https://qt-mirror.dannhauer.de']

### ISSUE THREAD

**Issue by bam80** [miurahr/aqtinstall#577]
```
$ ~/.local/bin/aqt list-tool linux desktop tools_qtcreator
ERROR   : Failed to download checksum for the file 'Updates.xml' from mirrors '['https://qt-mirror.dannhauer.de']
==============================Suggested follow-up:==============================
* Check your internet connection
* Consider modifying `requests.max_retries_to_retrieve_hash` in settings.ini
* Consider modifying `mirrors.trusted_mirrors` in settings.ini (see https://aqtinstall.readthedocs.io/en/stable/configuration.html#configuration)
* Please use 'aqt list-tool linux desktop' to check what tools are available.
```
The error might be caused by IP blocking from https://download.qt.io so I swapped it to https://qt-mirror.dannhauer.de mirror in `trusted_mirrors` setting.
Still it didn't help.

**Comment by miurahr** [comment]
Your mirror does not provide SHA256 checksum, so it cannot be trusted.

Sha256 sum is got like as

https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/tools_qtcreator/Updates.xml.sha256


Does your mirror

https://qt-mirror.dannhauer.de/online/qtsdkrepository/linux_x64/desktop/tools_qtcreator/Updates.xml.sha256

returns sha256 properly?

An issue is duplicated with https://github.com/miurahr/aqtinstall/issues/521

**Comment by bam80** [comment]
> Does your mirror returns sha256 properly?

Thanks, but if I read #521 right, no mirrors other than qt.io do?

**Comment by silven** [comment]
Even qt.io does not provide a .sha256 for all tools. For example tools_ifw is missing: https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/tools_ifw/Updates.xml.sha256

**Comment by sithlord48** [comment]
The following tools are missing this sha256 file (tested linux) 
 - tools_ifw
 - tools_maintenance
 - tools_maintenance_update_reminder

**Comment by ddalcino** [comment]
@silven
> Even qt.io does not provide a .sha256 for all tools. For example tools_ifw is missing: https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/tools_ifw/Updates.xml.sha256

Check the link again; it looks like it's working now.

@sithlord48
> The following tools are missing the sha256 file
> 
> * tools_ifw
> * tools_maintenance
> * tools_maintenance_update_reminder

I just checked for sha256 files for all of these; none of them are missing. You were correct that these files were missing at the time you wrote this comment though.

Notice that all of these files are dated with today's date. I have noticed a pattern where for the first couple of hours after a file is released on download.qt.io, the sha256 hashes are missing (for example https://github.com/miurahr/aqtinstall/issues/521#issuecomment-1248477537). I do not understand why there's a delay between the upload of the binary/xml files and their sha256 checksums, or how long the delay is. I have not observed every one of these delays, but the delays that i have observed have all been under 24 hours.
